xconn/codec.py

```diff
@@ -0,0 +1,16 @@
+from typing import Any, Generic, Type, TypeVar
+
+T = TypeVar("T")
+
+
+class Codec(Generic[T]):
+    def name(self) -> str:
+        raise NotImplementedError
+
+    def encode(self, obj: Any) -> bytes | str:
+        """Serialize a Python object to bytes."""
+        raise NotImplementedError
+
+    def decode(self, data: bytes | str, out_type: Type[T]) -> T:
```

Comment:
```suggestion
    def decode(self, data: Invocation | str, out_type: Type[T]) -> T:
```

We actually need a new class that essentially has args, kwargs and details. but is "neutral" to cover Event/Invocation